src/components/Questions/QuestionFile.vue

```diff
@@ -0,0 +1,304 @@
+import NcLoadingIcon from '@nextcloud/vue/dist/Components/NcLoadingIcon.js'
+import OcsResponse2Data from '../../utils/OcsResponse2Data.js'
+import QuestionMixin from '../../mixins/QuestionMixin.js'
+import axios from '@nextcloud/axios'
+import fileTypes from '../../models/FileTypes.js'
+import logger from '../../utils/Logger.js'
+import { generateOcsUrl } from '@nextcloud/router'
+import { loadState } from '@nextcloud/initial-state'
+import { showError } from '@nextcloud/dialogs'
+
+const FILE_SIZE_UNITS = {
+	kb: 1024,
+	mb: 1024 ** 2,
+	gb: 1024 ** 3,
+}
+
+export default {
+	name: 'QuestionFile',
+	components: {
+		IconChevronLeft,
+		IconDelete,
+		IconFileDocumentAlert,
+		NcActionButton,
+		NcActionCheckbox,
+		NcActionInput,
+		NcActionSeparator,
+		NcActionText,
+		NcLoadingIcon,
+	},
+
+	mixins: [QuestionMixin],
+	data() {
+		return {
+			fileTypes,
+			fileLoading: false,
+			maxFileSizeUnit: Object.keys(FILE_SIZE_UNITS)[0],
+			maxFileSizeValue: '',
+			allowedFileTypesDialogOpened: false,
+		}
+	},
+
+	computed: {
+		availableUnits() {
+			return Object.keys(FILE_SIZE_UNITS)
+		},
+		maxAllowedFilesCount() {
+			return this.extraSettings?.maxAllowedFilesCount || 1
+		},
+		allowedFileTypesLabel() {
+			const allowedFileTypes = []
+			if (this.extraSettings?.allowedFileTypes?.length) {
+				allowedFileTypes.push(...this.extraSettings.allowedFileTypes.map(type => fileTypes[type].label))
+			}
+
+			if (this.extraSettings?.allowedFileExtensions?.length) {
+				allowedFileTypes.push(...this.extraSettings.allowedFileExtensions)
+			}
+
+			if (allowedFileTypes.length) {
+				return t('forms', 'Allowed file types: {fileTypes}.', { fileTypes: allowedFileTypes.join(', ') })
+			}
+
+			return t('forms', 'All file types are allowed.')
+		},
+	},
+	mounted() {
+		if (this.extraSettings.maxFileSize) {
+			Object.keys(FILE_SIZE_UNITS).forEach(unit => {
+				if (this.extraSettings.maxFileSize > FILE_SIZE_UNITS[unit]) {
+					this.maxFileSizeUnit = unit
+				}
+			})
+
+			this.maxFileSizeValue = this.extraSettings.maxFileSize / FILE_SIZE_UNITS[this.maxFileSizeUnit]
+		}
+	},
+
+	methods: {
+		async onFileInput() {
+			const fileInput = this.$refs.fileInput
+			const formData = new FormData();
+
+			[...fileInput.files].forEach(file => {
+				formData.append('files[]', file)
+			})
+
+			formData.append('shareHash', loadState('forms', 'shareHash'))
+
+			const url = generateOcsUrl('apps/forms/api/v2.5/uploadFiles/{formId}/{questionId}', {
+				formId: this.formId,
+				questionId: this.id,
+			})
+
+			let response
+			try {
+				this.fileLoading = true
+				response = await axios.post(url, formData, { headers: { 'Content-Type': 'multipart/form-data' } })
+			} catch (error) {
+				logger.error('Error while submitting the form', { error })
+				showError(t('forms', 'There was an error during submitting the file: {message}.',
+					{ message: error.response.data.ocs.meta.message }))
+
+				return
+			} finally {
```

Comment:
```suggestion
					{ message: error.response.data.ocs.meta.message }))
			} finally {
```

Is this `return` really necessary here?

Reply:
yes, because I don't want to execute next line:

```
this.$emit('update:values', [...this.values, ...OcsResponse2Data(response)])
```

if server responded with an error

Reply:
I didn't know that you can trigger the return after running finally coming from the catch block when you insert a return there.